code/src/microphone_controller.cpp

```diff
@@ -0,0 +1,40 @@
+#include "../headers/microphone_controller.hpp"
+
+uint16_t r2d2::microphone::microphone_controller_c::read_mic() {
+    if (!muted){
+        return r2d2::microphone::microphone_controller_c::mic.read(); // return adc pin (microphone) value
+    }
+    else{
+        return 0; // don't return an audible value
+    }
+}
+
+void r2d2::microphone::microphone_controller_c::sample(){
+    for (int i=0; i < 63; ++i){
```

Comment:
This 63 is hardcoded. Try and use the buffer_length you use in the module_c here

Reply:
The 63 is the internal buffer size for internal communication (in 16 bit chunks) as agreed upon with @itzandroidtab . However I did have this magic number replaced with a get_buffer_size() function which now makes it so the magic number is only in one place with commentary for why it is statically defined.